README.md

```diff
@@ -25,42 +28,74 @@ If you need to install docker compose, follow [these instructions](https://docs.
 ### Configure PostGIS
 Enable geospatial features by installing the postgis extension
 
-```
+```sh
 npm run pg:configure
 ```
 
 ### Drizzle ORM setup
 Structure the database tables with drizzle
 
-```
+```sh
 npm run drizzle:migrate
 ```
 
 ### Data loading
 Load the data into the tables. Under the hood, this uses the Postgres COPY command.
 
-```
+```sh
 npm run pg:copy
 ```
 
-### Run the project
-Finally, to run the project locally:
+## Local development
+The OpenAPI documentation and the API implementation complement each other. 
+The OpenAPI documentation is written first, defining expectations for endpoints.
+The API is subsequently implemented based on the OpenAPI documentation.
+
+### OpenAPI development
+The OpenAPI documentation is written in the `openapi/openapi.yaml` file.
+The [redoc](https://redocly.com/docs/redoc/) cli provides commands to manage the documentation,
+with a few key commands integrated into npm scripts.
+
+When writing OpenAPI documentation:
+- run `npm run redoc:dev`
+- open localhost:8080
+- make edits to `openapi/openapi.yaml`
+- view the updates in real time through port 8080
+
+To check validity of OpenAPI documentation:
+- run `npm run redoc:lint`
+
+### API development
+The API integrates with the OpenAPI documentation by implementing its endpoints. 
+The OpenAPI documentation defines schemas for the API to interact with. 
+The API uses [Kubb](https://www.kubb.dev) to automatically generate 
+Typescript definitions and zod schemas from the OpenAPI documentation.
+The generated code is kept in `/src/gen`. Never make manual changes to files in this folder.
+
+Generate these resources with:
+```sh
+npm run generate
```

Comment:
Can we also add that we get a 
```console
✖ Something went wrong

  Error: The "paths[0]" argument must be of type string. Received undefined
```
when we run this and we can safely ignore

Reply:
good catch. will add